ui/admin/app/routes/scopes/scope.js

```diff
@@ -33,17 +33,13 @@ export default class ScopesScopeRoute extends Route {
    * @param {string} params.scope_id
    * @return {Promise{ScopeModel}}
    */
-  model({ scope_id: id }) {
+  async model({ scope_id: id }) {
```

Comment:
Does this need to be `async` since we are not `awaiting` anything?

Reply:
Marked it as async since it returns a promise

Reply:
I think we only need `async` if we are explicitly using `await`. Otherwise, ember will handle the promise for us implicitly. For example, we if did `const response = await this.store.findRecord('scope', id)` then checked if the response contains an error, we would need to explicitly add `async`. I could be wrong and fully ok with that. Just making sure I know why `async` is needed. This [link](https://guides.emberjs.com/release/routing/specifying-a-routes-model/#toc_fetch-example) is what I am basing my assumption on.

Reply:
Yup ur correct, it's only explicitly needed when using await. For context, my thought process was that it was better practice to label functions that return a promise as async for clarity/readability, and so that the return value is always a promise but after looking at the link u posted and reading this [article](https://dev.to/somedood/best-practices-for-es2017-asynchronous-functions-async-await-39ji) I'm leaning towards removing it to avoid unnecessary wrapping.

Reply:
I think we discussed this before but you don't _need_ async if you're just returning a promise but I think you should. I'm of the opinion that if you're returning a promise, you should always mark the function as async for readability and to make it obvious the function is asynchronous. It also guarantees any return is a promise so you don't accidentally return a normal value and a promise in the same function. 

There shouldn't be any downside, there isn't any unnecessary wrapping since you're just returning a promise regardless, the unnecessary wrapping would occur if you returned an awaited promise (i.e. `return await asyncFunction()`)

Reply:
That makes since for documentation purposes. In this case, I think we can lean on the doc comment `@return {Promise{ScopeModel}}` to make it clear for future engineers.